feat: Add timeouts block to cortex

chris-codaio opened the pull request:
These services can take well over 30 minutes to create and sometimes many hours.  Add a timeout block so consumers can extend the default 30 minute timeout.

Patterned after the timeouts block on the existing `tag_association` object.

PTAL @sfc-gh-asawicki @sfc-gh-mloring

sfc-gh-asawicki commented on docs/resources/cortex_search_service.md:
```diff
@@ -77,6 +75,16 @@ resource "snowflake_cortex_search_service" "test" {
+
+Optional:
+
+- `create` (String)
```
The changes in this file seem to have been added manually. Please revert them and run `make pre-push` following the contribution guidelines: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CONTRIBUTING.md#before-submitting-the-pr (because this file is generated).

chris-codaio replied:
Done - ran `make pre-push` which updated these docs.

sfc-gh-asawicki commented:
Hey @chris-codaio. Thanks for adding the timeouts. I have left just one comment about the changes in the generated file. Otherwise LGTM.

chris-codaio commented:
> Hey @chris-codaio. Thanks for adding the timeouts. I have left just one comment about the changes in the generated file. Otherwise LGTM.

I believe I've addressed your comment.  PTAL when you get a chance!
Do you know when this might hit the release channel so we can plan accordingly on our end?

sfc-gh-asawicki commented:
Hey @chris-codaio. Thanks again for the contribution. It will be a part of the next release (tomorrow or on Friday).
